src/Microsoft.AspNet.Http.Core/Extensions/MapWhenMiddleware.cs

```diff
@@ -20,27 +20,13 @@ public MapWhenMiddleware([NotNull] RequestDelegate next, [NotNull] MapWhenOption
 
         public async Task Invoke([NotNull] HttpContext context)
         {
-            if (_options.Predicate != null)
+            if (_options.Predicate(context))
```

Comment:
With effort. They would have to do `app.Use(next => new MapWhenMiddleware(next, options).Invoke);` since we don't provide an extension that takes the options directly.